projects/storefrontlib/src/cms-components/cart/cart-shared/cart-item/cart-item.component.ts

```diff
@@ -53,6 +61,26 @@ export class CartItemComponent implements OnInit {
     );
   }
 
+  ngOnChanges(changes: SimpleChanges) {
+    this.populateCartItemContext(changes);
+  }
+
+  private populateCartItemContext(changes: SimpleChanges) {
+    if (this.cartItemContext) {
+      const newChunk = Object.entries(changes).reduce(
+        (acc, [key, change]) => ({ ...acc, [key]: change.currentValue }),
+        {} as CartItemContextModel
+      );
+
+      let oldChunk: CartItemContextModel;
+      this.cartItemContext.context$
+        .subscribe((val) => (oldChunk = val ?? {}))
+        .unsubscribe();
+
+      this.cartItemContext['context$$'].next({ ...oldChunk, ...newChunk });
```

Comment:
Let's hope we can do better here ;-)

Reply:
Would you mind if we do simpler:

```typescript
class CartItemContext {
  context$: Observable<...> = new BehaviorSubject<...>({});
  // note: no context$$
```
and push next values in the component:
```typescript
const context$ = this.cartItemContext.context$ as BehaviorSubject<...>;
context$.next({...context$.value, ...newChunk})
```

Reply:
like it much better ;-)